tensorflow_datasets/sequence/quickdraw.py

```diff
@@ -0,0 +1,123 @@
+                "own .npz file, for example, cat.npz."
+            ),
+            features=tfds.features.FeaturesDict(
+                {
+                    "strokes": tfds.features.Tensor(shape=(None, 3), dtype=tf.int16),
+                    "label": tfds.features.ClassLabel(names_file=labels_path),
+                }
+            ),
+            supervised_keys=("strokes", "label"),
+            urls=["https://github.com/googlecreativelab/quickdraw-dataset"],
+            citation=(
+                "A Neural Representation of Sketch Drawings, "
+                "D. Ha and D. Eck, arXiv:1704.03477v4, 2017."
+            ),
+        )
+
+    def _split_generators(self, dl_manager):
+        # The QuickDraw bitmap repository is structured as one .npz file per label
+        # containing the three splits.
+        labels = self.info.features["label"].names
+        urls = {
+            label: "{}/sketchrnn/{}.npz".format(_QUICKDRAW_BASE_URL, label)
+            for label in labels
+        }
+
+        file_paths = dl_manager.download(urls)
+
+        # Prepare the destinations used to unpack the split
+        extract_dir = dl_manager._extract_dir
+        for label in tqdm.tqdm(file_paths, desc="Unpacking downloaded archives."):
+            data = np.load(file_paths[label], encoding="latin1")
+            for split in ["train", "test", "valid"]:
+                split_dir = os.path.join(extract_dir, split)
+                if not tf.io.gfile.exists(split_dir):
+                    tf.io.gfile.makedirs(split_dir)
+                np.save(os.path.join(split_dir, label), data[split])
+                assert os.path.exists(os.path.join(split_dir, "{}.npy".format(label)))
+
+        return [
+            tfds.core.SplitGenerator(
+                name=tfds.Split.TRAIN,
+                num_shards=20,
+                gen_kwargs={"file_paths": os.path.join(extract_dir, "train")},
+            ),
+            tfds.core.SplitGenerator(
+                name=tfds.Split.TEST,
+                num_shards=5,
+                gen_kwargs={"file_paths": os.path.join(extract_dir, "test")},
+            ),
+            tfds.core.SplitGenerator(
+                name=tfds.Split.VALIDATION,
+                num_shards=5,
+                gen_kwargs={"file_paths": os.path.join(extract_dir, "valid")},
+            ),
+        ]
+
+    def _generate_examples(self, file_paths):
+        """Generate QuickDraw bitmap examples.
+
+        Given the path of the folder containing all the class data
+
+        Args:
+            file_paths: (dict of {str: str}) the paths to files containing the data.
+
+        Yields:
+            The QuickDraw examples, as defined in the dataset info features.
+
+        """
+        for path in tf.io.gfile.listdir(file_paths):
+            data = np.load(os.path.join(file_paths, path))
+            for strokes in data:
+                yield {"strokes": strokes, "label": path.strip(".npy")}
```

Comment:
I totally brain-farted on that. Changing it immediately.

Reply:
Gotta love 1-character bug fixes. The letter `r` is not that common in that category, though. ;-)

Reply:
Apparently `.rstrip` is wrong as well. In case of `spoon.npy` it will output `spoo`. In the end good old [-4] should do the trick. I will finish testing and push an updated version containing both the dataset in stroke-5 and stroke-3 format coupled with a slightly quicker unpacking phase thanks to multiprocessing.

Reply:
Oops, indeed! 👍 Note: `[:-4]` not `[-4]`